sqlglot/parser.py

```diff
@@ -226,6 +225,22 @@ def raise_error(self, message, token=None):
         if self.error_level == ErrorLevel.WARN:
             logging.error(self.error)
 
+    def expression(self, exp_class, **kwargs):
+        instance = exp_class(**kwargs)
+        self.validate(instance)
+        return instance
+
+    def validate(self, expression):
```

Comment:
[Nitpick/Optional] I would be more specific about what's being validated here and call it explicitly `validate_expression`.

Reply:
done